tests/sentry/workflow_engine/processors/test_action.py

```diff
@@ -58,10 +61,27 @@ def test_multiple_workflows(self):
         status_3.update(date_updated=timezone.now() - timedelta(days=2))
 
         triggered_actions = filter_recently_fired_workflow_actions(
-            set(DataConditionGroup.objects.all()), self.group
+            set(DataConditionGroup.objects.all()), self.event_data
         )
         assert set(triggered_actions) == {self.action, action_3}
 
         for status in [status_1, status_2, status_3]:
             status.refresh_from_db()
             assert status.date_updated == timezone.now()
+
+    def test_update_workflow_fire_histories(self):
+        WorkflowFireHistory.objects.create(
+            workflow=self.workflow, group=self.group, event_id=self.group_event.event_id
+        )
+
+        actions = Action.objects.all()
```

Comment:
This might read better if you explicitly pass in `has_fired_actions=False` when creating it. Even though it's the default, I had to go check the model definition to see what the difference between the created model and the one fetched after was. Also maybe add `assert actions.count() == 1` (I'm assuming that's the count?) so I can tell that there's a 1:1 with actions and workflowfirehistories with `has_fired_actions=True`